libexec/crkeng_stop_word_stat.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+"""
+find out the stop words in crkeng.xml's translations
+"""
+import argparse
+import re
+from collections import Counter
+from pprint import pprint
+from typing import List
+from xml.etree import ElementTree as ET
+
+parser = argparse.ArgumentParser(
+    description="find out the stop words in crkeng.xml's translations"
+)
+parser.add_argument("crkeng_file")
+
+if __name__ == "__main__":
+    word_pattern = re.compile(r"[\w.\-/']+")
+    args = parser.parse_args()
+    root = ET.parse(str(args.crkeng_file)).getroot()
+    elements = root.findall(".//t")
+    translations = {e.text.lower() for e in elements if e.text is not None}
+
+    words: List[str] = [
+        word
+        for translation in translations
+        for word in re.findall(word_pattern, translation)
+    ]
+    print("result is case insensitive:")
+    pprint(Counter(words).most_common(100))
+
+"""
+result on CreeDictionary/res/dictionaries/crkeng_cw_md_200314.xml 
+[('s/he', 17963),
+ ('is', 5010),
+ ('s.o.', 4665),
+ ('a', 4647),
+ ('it', 4255),
+ ('s.t.', 3385),
+ ('he', 2605),
+ ('the', 2566),
+ ('of', 2090),
+ ('to', 2030),
+ ('in', 2022),
+ ('has', 1464),
+ ('for', 1394),
+ ('things', 1233),
+ ('with', 1228),
+ ('by', 1201),
+ ('or', 1136),
+ ('people', 1083),
+ ('on', 1064),
+ ('makes', 999),
+ ('e.g.', 867),
+ ('as', 859),
+ ('his/her', 712),
```

Comment:
@eddieantonio @aarppe I wrote a script that does statistics about stop words. Currently only five are used. Do you think we can be less conservative and use more than five of them?

Reply:
I'd prefer a search index with tf.idf or Okapi BM25 instead of stop words, as it takes into account these statistics automatically.

Reply:
sounds cool!

Reply:
For future reference, the short stop word list was motivated by ruling out articles/determiners `a`, `an`, `the` and the most formulaic of CW conventions, i.e. `s/he`,` s.o.` and `s.t.`. The motivation was the users might have good reasons to search with common prepositions like `of` and `with`.

On the longer term, such elements which are grammatical in English and which do not always have a correspondent in Cree might be replaced with a "grammar entry", explaining how the grammatical function conveyed by the English preposition (or the like) can be (morpho-syntactically) communicated in Cree (for lack of a proper matching word).